hawkbit-dmf/hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/amqp/DelegatingAmqpErrorHandlerTest.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.amqp;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.util.ErrorHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Feature("Unit Tests - Delegating Conditional Error Handler")
+@Story("Delegating Conditional Error Handler")
+public class DelegatingAmqpErrorHandlerTest {
+
+    @Test
+    @Description("Verifies that with a list of conditional error handlers, the error is delegated to specific handler.")
+    public void verifyDelegationHandling(){
+        List<AmqpErrorHandler> handlers = new ArrayList<>();
+        handlers.add(new IllegalArgumentExceptionHandler());
+        handlers.add(new IndexOutOfBoundsExceptionHandler());
+        Assertions.assertThrows(IllegalArgumentException.class,
```

Comment:
for the consistency we agreed upon using assertJ for exception handling, could you please adapt?